shared/src/business/useCases/practitioners/editPractitionerDocumentInteractor.test.ts

```diff
@@ -0,0 +1,68 @@
+import { InvalidEntityError, UnauthorizedError } from '../../../errors/errors';
+import { ROLES } from '../../entities/EntityConstants';
+import { applicationContext } from '../../test/createTestApplicationContext';
+import { editPractitionerDocumentInteractor } from './editPractitionerDocumentInteractor';
+
+describe('editPractitionerDocumentInteractor', () => {
+  const testUser = {
+    role: ROLES.admissionsClerk,
+    userId: 'admissionsclerk',
+  };
+  const barNumber = 'PT4785';
+  const mockDocumentMetadata = {
+    categoryName: 'Application',
+    categoryType: 'Application',
+    description: 'bird is the word',
+    fileName: 'testFile.pdf',
+    location: undefined,
+    practitionerDocumentFileId: 'ad69486a-2489-4eb1-bee6-bdb6cc0257b4',
+  } as any;
+
+  beforeAll(() => {
+    applicationContext.getCurrentUser.mockReturnValue(testUser);
+  });
+
+  it('should throw an unauthorized error when the user does not have permission to update the practitioner user', async () => {
+    const testPetitionerUser = {
+      role: ROLES.petitioner,
+      userId: 'petitioner',
+    };
+    applicationContext.getCurrentUser.mockReturnValueOnce(testPetitionerUser);
+
+    await expect(
+      editPractitionerDocumentInteractor(applicationContext, {
+        barNumber,
+        documentMetadata: mockDocumentMetadata,
+      }),
+    ).rejects.toThrow(UnauthorizedError);
+  });
+
+  it('should throw a validation error if the practitioner document has the wrong type', async () => {
+    await expect(
+      editPractitionerDocumentInteractor(applicationContext, {
+        barNumber,
+        documentMetadata: { ...mockDocumentMetadata, categoryType: 'GG' },
+      }),
+    ).rejects.toThrow(InvalidEntityError);
+  });
+
+  it('should throw a validation error if the practitioner document is missing information', async () => {
```

Comment:
How would file name be undefined from an edit standpoint?  You changed the stateDrivenFileInput to default to existing filename.

Reply:
This is only when editing the document- when no document has been uploaded, `fileName` is undefined, but `existingFileName` has been set on the form